include/oneapi/tbb/detail/_concurrent_queue_base.h

```diff
@@ -189,7 +192,7 @@ class micro_queue {
                 k + queue_rep_type::n_queue, index == items_per_page - 1 ? p : nullptr );
             if (p->mask.load(std::memory_order_relaxed) & (std::uintptr_t(1) << index)) {
                 success = true;
-                assign_and_destroy_item( dst, *p, index );
+                select_pop_operation(dst, *p, index, do_assign);
```

Comment:
The function name doesn't correspond with actions inside function. What do you think about `pop_element`?

Reply:
Irrelevant for the newest version of the patch